src/tss2-fapi/api/Fapi_Delete.c

```diff
@@ -642,7 +642,8 @@ Fapi_Delete_Finish(
             fallthrough;
 
         statecase(context->state, ENTITY_DELETE_KEY_WAIT_FOR_AUTHORIZATION);
-            if (object->misc.key.persistent_handle) {
+            /* Delete persistent object if not prohibited. */
+            if (object->misc.key.persistent_handle && !object->misc.key.delete_prohibited) {
```

Comment:
IMHO; we should make this an explicit error message, like
"Failed to delete key because it was not created by the tss Feature API" or similar